spec/support/features/personal_key_helper.rb

```diff
@@ -27,11 +27,23 @@ def trigger_reset_password_and_click_email_link(email)
     click_email_link_matching(/reset_password_token/)
   end
 
-  def scrape_personal_key
-    new_personal_key_words = []
-    page.all(:css, '[data-personal-key]').each do |node|
-      new_personal_key_words << node.text
-    end
-    new_personal_key_words.join('-')
+  def scrape_personal_key(jumbled_for_entry: false)
+    code_segments = page.all(:css, '.separator-text__code').map(&:text)
+
+    return code_segments.join('-') if !jumbled_for_entry
+
+    # Include dash between some segments and not others
+    code = code_segments[0..1].join('-') + code_segments[2..3].join
+
+    # Randomize case
+    code = code.chars.map { |c| (rand 2) == 0 ? c.downcase : c.upcase }.join
+
+    # De-normalize Crockford encoding
+    code = code.sub('1', 'l').sub('0', 'O')
+
+    # Add extra characters
+    code += 'abc123qwerty'
+
+    code
```

Comment:
Personally, I would rather have one test scenario that jumbles the personal key like this, instead of complicating every single use of this helper method in all our tests?

That way if it ever fails, there's one test with a clear description like "it can handle a crockford-encoded translation swap" or something instead of every single test that touches a personal key failing?

Reply:
Fair point, I'll find another place for it. There was some jumbling already happening in `acknowledge_and_confirm_personal_key`, which itself is a shared helper. So maybe we need a new test case.

Reply:
Revised in 87b7c5d83.